### PR TITLE
Fix bugs in generated C++ for ports

### LINKED ISSUE #209: Fix bugs in generated C++ for ports
We need to fix the following bugs in the Python port autocoder:

1. The Python autocoder generates incorrect code in the `invoke` function of the `OutputPort` class. If port serialization is turned on, then there is an access to `m_port` guarded by a check that either `m_port` or `m_serPort` is non-null. In this case, if `m_serPort` is non-null and `m_port` is null, a seg fault occurs.

2. The Python autocoder assumes that `nullptr == 0`, which C++ does not guarantee.

See https://github.com/fprime-community/fpp/pull/208#issuecomment-1385933219.

### PR DESCRIPTION
**Pull request by tiffany1618**
Closes #209

### CODE REVIEW

File: compiler/lib/src/main/scala/codegen/CppWriter/PortCppWriter.scala [hunk 1]
```diff
@@ -501,38 +501,47 @@ case class PortCppWriter (
...
+      )
+      case None => List(
+        lines(
+          s"""|
+              |#if FW_PORT_SERIALIZATION
+              |FW_ASSERT((this->m_port != nullptr) || (this->m_serPort != nullptr));
+              |
+              |if (this->m_port != nullptr) {
+              |  $invokeCall
+              |} else {
```
Comment (bocchino) [inline compiler/lib/src/main/scala/codegen/CppWriter/PortCppWriter.scala:520]: Can we move the else statement to the next line?

**Review by bocchino — COMMENTED** [review]
Looks great! Just one comment.

File: compiler/lib/src/main/scala/codegen/CppWriter/PortCppWriter.scala (hunk 1 above)
Comment (tiffany1618) [reply compiler/lib/src/main/scala/codegen/CppWriter/PortCppWriter.scala]: Ok, fixed!

**Review by bocchino — APPROVED** [review]
Looks great!

**Pull request merged**